### PR TITLE
Added Azure.ContainerApp.ManagedIdentity

### LINKED ISSUE #2096: Managed Identity should be enabled for Container Apps
### Existing rule

_No response_

### Suggested rule

Managed Identity should be enabled for Container Apps.

### Pillar

Security

### Additional context

- [Use identity-based authentication](https://learn.microsoft.com/azure/architecture/framework/security/design-identity-authentication#use-identity-based-authentication)
- [Managed identities in Azure Container Apps](https://learn.microsoft.com/azure/container-apps/managed-identity)
- [Azure deployment reference](https://learn.microsoft.com/azure/templates/microsoft.app/containerapps#managedserviceidentity)

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Fixes #2096 

Added Azure.ContainerApp.ManagedIdentity rule.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

File: tests/PSRule.Rules.Azure.Tests/Azure.ContainerApp.Tests.ps1 [hunk 1]
```diff
@@ -52,5 +52,22 @@ Describe 'Azure.ContainerApp' -Tag 'ContainerApp' {
             $ruleResult.Length | Should -Be 1;
             $ruleResult.TargetName | Should -BeIn 'capp-A';
         }
```
Comment (BernieWhite) [inline tests/PSRule.Rules.Azure.Tests/Azure.ContainerApp.Tests.ps1:54]: You may need to bump `Azure.ContainerApp.Insecure` for the new resource instances.

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Nice work @BenjaminEngeset! Just need to tweak `Azure.ContainerApp.Insecure` test, otherwise all good.

**Comment by BenjaminEngeset** [comment]
@BernieWhite 

Check latest updates :)

**Review by BernieWhite — APPROVED** [review]
Thanks @BenjaminEngeset looks good. All good to merge.

**Pull request merged**